src/lib/components/ui/MenuButton/MenuButton.jsx

```diff
@@ -0,0 +1,128 @@
+        onClick={() => setIsExpanded(!isExpanded)}
+        ref={props.forwardedRef}
+        type="button"
+      >
+        {props.beforeLabel && (
+          <span>
+            {props.beforeLabel}
+          </span>
+        )}
+        {props.label}
+        {props.afterLabel && (
+          <span>
+            {props.afterLabel}
+          </span>
+        )}
+        {props.loadingIcon && (
+          <span>
+            {props.loadingIcon}
+          </span>
+        )}
+      </button>
+      {isExpanded && !props.disabled && (
+        <div id={props.id ? `${props.id}__expandedMenuContainer` : undefined}>
+          {props.buttons.map((button) => {
+            if (button.type === 'separator') {
+              return <hr />;
+            }
+
+            return (
+              <button
+                disabled={button.disabled || !!button.loadingIcon || props.disabled}
+                id={button.id || undefined}
+                onClick={(e) => {
+                  const handlerResult = button.clickHandler(e);
+
+                  if (handlerResult instanceof Promise) {
+                    return handlerResult.then((response) => {
+                      setIsExpanded(false);
+                      return response;
+                    });
+                  }
+
+                  setIsExpanded(false);
+                  return handlerResult;
+                }}
+                ref={button.forwardedRef}
+                type="button"
+              >
+                {button.beforeLabel && (
+                  <span>
+                    {button.beforeLabel}
+                  </span>
+                )}
+                {button.label}
+                {button.afterLabel && (
+                  <span>
+                    {button.afterLabel}
+                  </span>
+                )}
+                {button.loadingIcon && (
+                  <span>
+                    {button.loadingIcon}
+                  </span>
+                )}
+              </button>
+            );
+          })}
+        </div>
+      )}
+    </div>
+  );
+};
+
+MenuButton.defaultProps = {
+  afterLabel: null,
+  beforeLabel: null,
+  disabled: false,
+  forwardedRef: undefined,
+  id: undefined,
+  loadingIcon: null,
+  priority: 'default',
+  size: 'medium',
+  variant: 'primary',
+};
+
+MenuButton.propTypes = {
+  afterLabel: PropTypes.element,
+  beforeLabel: PropTypes.element,
+  buttons: PropTypes.arrayOf(PropTypes.oneOfType([
+    PropTypes.shape({
+      afterLabel: PropTypes.element,
+      beforeLabel: PropTypes.element,
+      clickHandler: PropTypes.func,
+      disabled: PropTypes.bool,
+      forwardedRef: PropTypes.func,
+      id: PropTypes.string,
+      label: PropTypes.string.isRequired,
+      loadingIcon: PropTypes.element,
+      type: PropTypes.oneOf(['button']),
+      variant: PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger', 'dark']),
```

Comment:
I wouldn't allow any of these, it's a dropdown menu after all. Maybe only `danger` variant could be useful for destructive actions and in a subtle way like red text or so.